Build configuration is not skipped when skipnative is presented as argument to build script (Linux)

When trying to build only mscorlib on Linux using the `skipnative` argument to the `build.sh` script, the build configuration step is not skipped (in method `build_coreclr`). Due to the condition in line 99 of `build.sh`, eventprovider is not generated, but is needed later during build configuration.

This results in the following error

```
prajwal@work $ ./build.sh arm skipnative skiptests
...
Config spew
...
-- Performing Test GETPWUID_R_SETS_ERRNO - Success
-- Performing Test FILE_OPS_CHECK_FERROR_OF_PREVIOUS_CALL
-- Performing Test FILE_OPS_CHECK_FERROR_OF_PREVIOUS_CALL - Failed
-- Performing Test UNWIND_CONTEXT_IS_UCONTEXT_T
-- Performing Test UNWIND_CONTEXT_IS_UCONTEXT_T - Success
CMake Error at src/pal/src/CMakeLists.txt:281 (add_subdirectory):
  The source directory

    /home/prajwal/dotnet/coreclr/coreclr/bin/obj/Linux.arm.Debug/Generated/eventprovider does not contain a CMakeLists.txt file.

-- Configuring incomplete, errors occurred!
```

The Windows build script skips the configuration step if only mscorlib is to be built. Do we need to configure the build even though actual building of native components is skipped later?

The fix would be to move the
```
if [ $__SkipCoreCLR == 1 ]; then
        echo "Skipping CoreCLR build."
        return
fi
```
condition to the beginning of the `build_coreclr` function.

--- a/build.sh
+++ b/build.sh
@@ -87,6 +87,11 @@ check_prereqs()
 build_coreclr()
 {
 
+    if [ $__SkipCoreCLR == 1 ]; then
+        echo "Skipping CoreCLR build."
+        return
+    fi
+
 # Event Logging Infrastructure
    __GeneratedIntermediate="$__IntermediatesDir/Generated"
    __GeneratedIntermediateEventProvider="$__GeneratedIntermediate/eventprovider_new"
@@ -161,11 +166,6 @@ build_coreclr()
         # Regenerate the CMake solution
         echo "Invoking \"$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh\" \"$__ProjectRoot\" $__ClangMajorVersion $__ClangMinorVersion $__BuildArch $__BuildType $__CodeCoverage $__IncludeTests $generator $__cmakeargs"
         "$__ProjectRoot/src/pal/tools/gen-buildsys-clang.sh" "$__ProjectRoot" $__ClangMajorVersion $__ClangMinorVersion $__BuildArch $__BuildType $__CodeCoverage $__IncludeTests $generator "$__cmakeargs"
-    fi
-
-    if [ $__SkipCoreCLR == 1 ]; then
-        echo "Skipping CoreCLR build."
-        return
     fi
 
     # Check that the makefiles were created.

